library/core/src/main/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelector.java

```diff
@@ -2535,4 +2494,59 @@ public int hashCode() {
 
   }
 
+  /** Represents how well an text track matches the selection {@link Parameters}. */
+  protected static final class TextTrackScore implements Comparable<TextTrackScore> {
+
+    private final boolean isWithinRendererCapabilities;
+    private final int preferredLanguageScore;
+    private final int localeLanguageMatchIndex;
+    private final int localeLanguageScore;
+    private final boolean isDefaultSelectionFlag;
+
+    public TextTrackScore(Format format, Parameters parameters, int formatSupport) {
+      isWithinRendererCapabilities = isSupported(formatSupport, false);
+      preferredLanguageScore = getFormatLanguageScore(format, parameters.preferredTextLanguage);
+      isDefaultSelectionFlag = (format.selectionFlags & C.SELECTION_FLAG_DEFAULT) != 0;
+      String[] localeLanguages = Util.getSystemLanguageCodes();
```

Comment:
Using System locale languages isn't part of the original logic. Can you leave that out and ensure that the logic stays the same?  The same is true for other parts of the logic that just disappeared with this change. For example selectUndeterminedTextLanguage isn't handled anymore and neither is the forced flag.

Reply:
My bad, I misunderstood what you were expecting. So I first re-introduced the existing score logic into TextTrackScore. However, I still need to move to a comparison logic, am I right?

Reply:
Yes, that was the idea. The code right now is correct and a step in the right direction, but the main idea was to change all the conditions and scores that are currently in the constructor of TextTrackScore to comparisons in TextTrackScore.compareTo. Because the comparisons are easier to maintain than the score numbers. And it becomes easier to add new features like the role flag parsing.

Reply:
Thanks for pointing the right direction.
I've almost done the move to comparisons in `TextTrackScore.compareTo` but I have one issue.
In the existing logic of `selectTextTrack`, it allows to avoid selecting a text track when none conditions matched above:
`          } else {
            // Track should not be selected.
            continue;
          }`

In my last commit, it always selects and enables the first TextTrack, also when not calling `.setPreferredTextLanguage()` on DefaultTrackSelector:

    Renderer:2 [
     Group:0, adaptive_supported=N/A [
       [X] Track:0, id=DVBSUB_TTML_fra_1947161=10000.track_id=30003, mimeType=application/ttml+xml, bitrate=10000, codecs=stpp, language=fra, supported=YES
     ]
     Group:1, adaptive_supported=N/A [
       [ ] Track:0, id=DVBSUB_TTML_fra_1947361=10000.track_id=30001, mimeType=application/ttml+xml, bitrate=10000, codecs=stpp, language=fra, supported=YES
     ]
   ]

So my implementation behaviour differs from the existing one right now.

Should I add a `TextTrackScore.shouldNotBeSelected` and test it in `selectTextTrack` to match the existing logic ?


          if (!trackScore.shouldNotBeSelected && ((selectedTrackScore == null) || trackScore.compareTo(selectedTrackScore) > 0)) {
            selectedGroup = trackGroup;
            selectedTrackIndex = trackIndex;
            selectedTrackScore = trackScore;
          }